File: packages/astro-polyglot/tests/fixtures/javascript/math_utils.js
/**
 * A simple math utility module with JSDoc annotations.
 * @module math-utils
 */

/**
 * Add two numbers.
 * @param {number} a - First operand.
 * @param {number} b - Second operand.
 * @returns {number} The sum of a and b.
 */
export function add(a, b) {
  return a + b;
}

/**
 * A calculator class with typed methods.
 * @class
 */
export class Calculator {
  /**
   * Add two integers.
   * @param {number} a - First operand.
   * @param {number} b - Second operand.
   * @returns {number} The sum.
   */
  add(a, b) {
    return a + b;
  }
}

File: packages/astro-polyglot/tests/fixtures/pascal/math_utils.pas
/**
 * @brief A simple math utility unit.
 */
unit MathUtils;

interface

  /**
   * @brief Add two integers.
   * @param a First operand.
   * @param b Second operand.
   * @return The sum.
   */
  function Add(a, b: Integer): Integer;

implementation

  function Add(a, b: Integer): Integer;
  begin
    Add := a + b;
  end;

end.
